### PR TITLE
[loki microservices]: allow custom gateway server configuration

### PR DESCRIPTION
**Pull request by irizzant**
Signed-off-by: irizzant <i.rizzante@gmail.com>

**What this PR does / why we need it**:

This PR allows users to specify custom server snippets to configure the gateway.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [X] Documentation added
- [X] Tests updated
- [X] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [X] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: production/ksonnet/loki/config.libsonnet [hunk 1]
```diff
@@ -42,6 +42,11 @@
 
     jaeger_reporter_max_queue: 1000,
 
+    // This is inserted into the gateway Nginx config file
+    // under the server directive
+
+    server_snippet: '',
```
Comment (kavirajk) [inline production/ksonnet/loki/config.libsonnet:48]: Couple of suggestions:
1. Please move `server_snippet` into `loki/gateway.libsonet` _config as it's used only on the nginx gateway now.
2. `server_snippet` -> `gateway_server_snippet` may be a better name?

**Review by kavirajk — COMMENTED** [review]
Like the idea, proposed couple of minor suggestions.

File: production/ksonnet/loki/config.libsonnet (hunk 1 above)
Comment (irizzant) [reply production/ksonnet/loki/config.libsonnet]: @kavirajk thanks for the hints, I updated gateway configuration accordingly

**Comment by irizzant** [comment]
@kavirajk I updated the code as requested, thanks

**Review by kavirajk — APPROVED** [review]
Thanks @irizzant LGTM :+1:

**Pull request merged**